packages/mwp-app-route-plugin/src/route.js

```diff
@@ -12,7 +15,13 @@ export const onPreResponse = {
 	 */
 	method: (request: HapiRequest, reply: HapiReply) => {
 		const response = request.response;
-
+		if (request.query) {
+			const trackingParam = request.query['_xtd']
+			if (trackingParam) {
+				response.header(EXTERNAL_TRACK_HEADER, trackingParam);
```

Comment:
I think this behavior might not actually be what you're hoping for - you're sending these headers back to the browser, but I would guess that the browser will just ignore them in the response? Are you intending to send them along with REST API requests instead? Is there another part of the ticket that add browser-side handling? The ticket and PR description are vague about how the headers are used.

Reply:
I will ask for a more clarified ticket description but the general idea is to allow for email tracking as stated in https://meetup.atlassian.net/wiki/spaces/DAT/pages/27754630/Email+Tracking

Reply:
No browser-handling

Reply:
ok, so `onPreResponse` modifies the response sent back to the browser, but the docs you linked indicate that the custom headers need to be sent to the REST API - the current implementation won't affect the requests to API.

Instead, you'll probably want to modify [this function](https://github.com/meetup/meetup-web-platform/blob/master/packages/mwp-api-proxy-plugin/src/util/send.js#L200) to inject custom headers into the REST API requests. One thing you'll have to find out from the data team is whether it's okay to add those headers to _all_ parallel REST API requests that correspond to opening an email or whether the headers should only be sent with one of the REST API requests - it's possible that the BE assumes that there will be only one REST request per application request.